src/webgpu/shader/execution/expression/binary/bitwise.spec.ts

```diff
@@ -26,63 +73,92 @@ Bitwise-or. Component-wise when T is a vector.
       .combine('type', ['i32', 'u32'] as const)
       .combine('inputSource', allInputSources)
       .combine('vectorize', [undefined, 2, 3, 4] as const)
-      .combine('compoundStmt', [false, true] as const)
   )
   .fn(async t => {
     const type = scalarType(t.params.type);
-    const V = t.params.type === 'i32' ? i32 : u32;
-    const cases = [
-      // Static patterns
-      {
-        input: [V(0b00000000000000000000000000000000), V(0b00000000000000000000000000000000)],
-        expected: V(0b00000000000000000000000000000000),
-      },
-      {
-        input: [V(0b11111111111111111111111111111111), V(0b00000000000000000000000000000000)],
-        expected: V(0b11111111111111111111111111111111),
-      },
-      {
-        input: [V(0b00000000000000000000000000000000), V(0b11111111111111111111111111111111)],
-        expected: V(0b11111111111111111111111111111111),
-      },
-      {
-        input: [V(0b11111111111111111111111111111111), V(0b11111111111111111111111111111111)],
-        expected: V(0b11111111111111111111111111111111),
-      },
-      {
-        input: [V(0b10100100010010100100010010100100), V(0b00000000000000000000000000000000)],
-        expected: V(0b10100100010010100100010010100100),
-      },
-      {
-        input: [V(0b00000000000000000000000000000000), V(0b10100100010010100100010010100100)],
-        expected: V(0b10100100010010100100010010100100),
-      },
-      {
-        input: [V(0b01010010001001010010001001010010), V(0b10100100010010100100010010100100)],
-        expected: V(0b11110110011011110110011011110110),
-      },
-    ];
-    // Permute all combinations of a single bit being set for the LHS and RHS
-    for (let i = 0; i < 32; i++) {
-      const lhs = 1 << i;
-      for (let j = 0; j < 32; j++) {
-        const rhs = 1 << j;
-        cases.push({
-          input: [V(lhs), V(rhs)],
-          expected: V(lhs | rhs),
-        });
-      }
-    }
-    await run(
-      t,
-      t.params.compoundStmt ? compoundBinary('|') : binary('|'),
-      [type, type],
-      type,
-      t.params,
-      cases
-    );
+    const cases = makeBitwiseOrCases(t.params.type);
+
+    await run(t, binary('|'), [type, type], type, t.params, cases);
```

Comment:
Might make sense to change `'|'` to `'|='` now, and remove the bodge in `binary()`

Reply:
Will do as a followup as I'll need to change all the `compoundBinary` to pass the extra `=`.